Sources/generate-swift-syntax-builder/gyb_syntax_support/AttributeKinds.py

```diff
@@ -0,0 +1,871 @@
+from .Child import Child
+from .Node import Node  # noqa: I201
+
+class Attribute(object):
+    def __init__(self, name, swift_name=None):
+        self.name = name
+        self.swift_name = swift_name or name
+
+    def macro_value(self):
+        return ""
+
+class TypeAttribute(Attribute):
+    def __init__(self, name):
+        super().__init__(name, name)
+
+    def macro_value(self):
+        return "TYPE_ATTR(" + self.name + ")"
```

Comment:
I assume that you are using `macro_value` from a `.def.gyb` file. It’s a little unfortunate that we currently spoil the SwiftSyntax repo with C++ macro logic. I understand that it’s easier to implement this way but maybe the cleaner separation of concerns would be to move this code out of the swift-syntax repo and have a big `if` statement in `.def.gyb` like the following:

```python
% if attr is TypeAttribute:
  TYPE_ATTR(${self.name})
% elif attr is DeclAttribute:
  …
% else:
%   fatal_error("Unhandled attribute class")
```

(realized that we need to handle subclasses before superclasses in the `if` after writing it, but that should be just a matter of the `if` order)